hnn_core/params.py

```diff
@@ -25,49 +25,44 @@ def _count_evoked_inputs(d):
     return nprox, ndist
 
 
-def _read_json(fname):
+def _read_json(param_data):
```

Comment:
Was this changed to work with GUI loading functions?

Reply:
I think this was just to clean up the code so that only `read_params()` receives the `fname` string to read the file contents, and then it sends the loaded data structure to either `_read_json()` or `_read_legacy_params()` for parsing the various. Perhaps we should consider renaming these functions (`_read_json()` and `_read_legacy_params()`) to say "parse" instead of "read" so that it's clear where I/O is actually happening now.

Reply:
The issue is that the ipywidgets file uploader reads the contents of the file and returns the binary string, so we need an intermediate function that can take the file contents and parse it, not just read from the file name.